core/common/src/main/java/alluxio/security/authentication/AbstractSaslClientHandler.java

```diff
@@ -47,7 +47,10 @@ public AbstractSaslClientHandler(ChannelAuthenticationScheme authScheme) {
   }
 
   @Override
-  public SaslMessage handleMessage(SaslMessage message) throws SaslException {
+  public synchronized SaslMessage handleMessage(SaslMessage message) throws SaslException {
```

Comment:
can you leave a comment as to why this and others have to be synchronized? (seems like it is for `mSaslClient`?)